game-core/src/main/java/games/strategy/engine/chat/ChatPlayerPanel.java

```diff
@@ -238,24 +239,11 @@ private String getDisplayString(final ChatParticipant chatParticipant) {
       return "";
     }
 
-    String extra = chatParticipant.isModerator() ? " " + TAG_MODERATOR : "";
-    String status = chat.getStatus(chatParticipant.getPlayerName());
+    final String extra = chatParticipant.isModerator() ? " " + TAG_MODERATOR : "";
+    final String status = Ascii.truncate(chat.getStatus(chatParticipant.getPlayerName()), 25, "");
+    final String suffix = status.isEmpty() ? "" : " (" + status + ")";
 
-    final StringBuilder sb = new StringBuilder();
-    if (status.length() > 0) {
-      if (status.length() > 25) {
-        status = status.substring(0, 25);
-      }
-      for (int i = 0; i < status.length(); i++) {
-        final char c = status.charAt(i);
-        if (c >= '\u0300' && c <= '\u036F') { // skip combining characters
-          continue;
-        }
```

Comment:
Note that this PR drops this replacement.
However if there are any objections to this, I can just revert the latest commit to get a version that keeps this replacement.

Reply:
We should understand it before dropping it; hopefully there was a good reason it was added to begin with. It's really annoying to see code like this, at least there was some commentary noted to help, but the exact logic is elusive.

Reply:
Well I checked the unicode table of this particular range and it's like the comment states:
All of those characters are "combination chars" that give a ^ to a character. According to wikipedia those are called [diacritic signs](https://en.wikipedia.org/wiki/Diacritic) as a whole.

So removing those should really not have a noticeable effect.